### ISSUE TITLE
[FEATURE] Add more ways to customize stateful testing

### ISSUE THREAD

**Issue by Stranger6667** [schemathesis/schemathesis#1144]
**Is your feature request related to a problem? Please describe.**
The current implementation of stateful testing via state machines always adds a random data source, but it may not be desired.

**Describe the solution you'd like**
Extend the `as_state_machine()` method so the end-user can disable the random data source at all.

**Additional context**
Comes from [a gitter thread](https://gitter.im/schemathesis/schemathesis?at=60865a78d7953918617ea39a).

**Comment by DavidFarago** [comment]
I would like to customize stateful testing to only report 500 errors, and focus on specific endpoints `^/weekly-supply`. Is this possible?

What I tried:
To focus on `^/weekly-supply`, I switched from `TestAPI = APIWorkflow.TestCase` to
```
@pytest.fixture
def state_machine():
    return schema.as_state_machine()

@schema.parametrize(endpoint="^/weekly-supply")
@settings(deadline=2000, max_examples=10)
def test_statefully(state_machine):
    run_state_machine_as_test(
        state_machine,
    )
```
which ran but did not lead to more focused testing in schemathesis < 3.7.0. 

With schemathesis 3.7.0, I get
```
E           InvalidSchema: test_statefully() got an unexpected keyword argument 'case', from `case=get_case_strategy(operation=APIOperation(path='/weekly-supply', method='put', schema=OpenApi30 for Foo Scheduling Server (1.0.0), verbose_name='PUT /weekly-supply', app=None, base_url='http://127.0.0.1:8088', path_parameters=ParameterSet(items=[]), headers=ParameterSet(items=[]), cookies=ParameterSet(items=[]), query=ParameterSet(items=[]), body=PayloadAlternatives(items=[OpenAPI30Body(definition={'schema': {'type': 'object', 'additionalProperties': False, 'properties': {'week': {'type': 'string', 'pattern': '\\A2021\\-W[0-5][0-9]\\Z'}, 'material': {'type': 'array', 'items': {'type': 'object', 'additionalProperties': False, 'properties': {'id': {'type': 'integer'}, 'quantity': {'type': 'integer'}, 'company_key': {'type': 'string'}}, 'required': ['id', 'quantity']}}}, 'required': ['week', 'material']}}, media_type='application/json', required=True)]), case_cls=<class 'schemathesis.models.Case'>), hooks=HookDispatcher(scope=<HookScope.TEST: 3>, _hooks=defaultdict(<class 'list'>, {})))` in @given
```

**Comment by Stranger6667** [comment]
Thanks for reporting!

indeed, it should work the way you tried. I’ll check it deeper today

**Comment by Stranger6667** [comment]
Btw, what Hypothesis version do you use?

**Comment by DavidFarago** [comment]
I used 
```
>>> hypothesis.__version__
'6.10.1'
```
Switching to 
```
>>> hypothesis.__version__
'6.13.4'
```
did not help.

**Comment by Stranger6667** [comment]
Here are some corrections to your example above:

```python
@pytest.fixture
def state_machine():
    return schema.clone(endpoint="^/weekly-supply").as_state_machine()

def test_statefully(state_machine):
    run_state_machine_as_test(
        state_machine,
        settings(deadline=2000, max_examples=10),
    )
```

> I would like to customize stateful testing to only report 500 errors, and focus on specific endpoints ^/weekly-supply. Is this possible?

Sure :)

```python
... # other imports are edited for brevity
from schemathesis.checks import not_a_server_error

@pytest.fixture
def state_machine():

    class APIWorkflow(schema.clone(endpoint="^/weekly-supply").as_state_machine()):
        def validate_response(self, response, case):
            case.validate_response(response, checks=(not_a_server_error,))

    return APIWorkflow

def test_statefully(state_machine):
    run_state_machine_as_test(
        state_machine,
        settings(deadline=2000, max_examples=10),
    )
```

Let me know if it solves your case :)

**Comment by DavidFarago** [comment]
Yes it did, thank you very much.

**Comment by joka** [comment]
Very helpfull example.

**Comment by Stranger6667** [comment]
@juk80x

Could you, please, elaborate on what kind of benefits you have in mind?

I am going to close this issue as the original problem was addressed in #1669 (which I think makes a lot of sense to have)
